src/snowflake/snowpark/dataframe_reader.py

```diff
@@ -43,6 +43,20 @@
 
 logger = getLogger(__name__)
 
+# We allow some common aliases to be used
+OPTION_ALIASES = {
+    "HEADER": ("SKIP_HEADER", lambda val: 1 if val else 0),
+    "DELIMITER": ("FIELD_DELIMITER", lambda val: val),
+    "SEP": ("FIELD_DELIMITER", lambda val: val),
+    "LINESEP": ("RECORD_DELIMITER", lambda val: val),
+    "PATHGLOBFILTER": ("PATTERN", lambda val: val),
+    "QUOTE": ("FIELD_OPTIONALLY_ENCLOSED_BY", lambda val: val),
+    "NULLVALUE": ("NULL_IF", lambda val: val),
+    "DATEFORMAT": ("DATE_FORMAT", lambda val: val),
+    "TIMESTAMPFORMAT": ("TIMESTAMP_FORMAT", lambda val: val),
+    "INFERSCHEMA": ("INFER_SCHEMA", lambda val: val),
+}
```

Comment:
@sfc-gh-yuliu If you could review this section

Reply:
@sfc-gh-yixie just for context. These aliases are mostly for compatibility in migration scenarios. these keywords are valid in spark and we just want to avoid friction when migrating